Client.php

```diff
@@ -84,11 +84,16 @@ class Client extends Connection
      */
     public function __construct(
         Socket\Client $client,
-        $endPoint               = '/',
+        $endPoint               = null,
         Http\Response $response = null
     ) {
         parent::__construct($client);
-        $this->setEndPoint($endPoint);
+
+        $this->setEndPoint(is_null($endPoint)?'/':$endPoint);
+        if( is_null($endPoint) &&
+            $client->getSocket() instanceof WebsocketSocket ) {
+            $this->setEndPoint($client->getSocket()->getEndPoint());
+        }
```

Comment:
We are testing `$endPoint` twice. Maybe the following flow is more appropriated:
```php
if (null === $endPoint) {
    $endPoint = '/';

    if ($client->getSocket() instanceof …) {
        $endPoint = $client->getSocket()->getEndPoint();
    }

    $this->setEndPoint($endPoint);
}
```
I might be wrong. Thoughts?